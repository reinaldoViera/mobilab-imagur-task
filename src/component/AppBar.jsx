import React from 'react'
import { AppBar, Toolbar, Typography, withStyles, IconButton } from '@material-ui/core';
import { withRouter } from "react-router";
import ArrowBack from '@material-ui/icons/ArrowBack';

const title = (path) => {
    switch (path) {
        case '/':
            return 'Gallery';
        default:
            return 'Album';
    }
}

const styles = (theme) => ({
    appBar: {
        zIndex: theme.zIndex.drawer + 1,
        transition: theme.transitions.create(['width', 'margin'], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen,
        }),
    },
});

export function MyAppBar({ classes, location, history }) {

    return (
        <div>
            <AppBar position="fixed" color="primary" className={classes.appBar}>
                <Toolbar>
                    {
                        location.pathname !== '/' &&
                        <IconButton onClick={() => history.goBack()} color="inherit" aria-label="Menu">
                            <ArrowBack />
                        </IconButton>
                    }
                    <Typography variant="h6" color="inherit">
                        {title(location.pathname)}
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    )
}

export default withStyles(styles)(withRouter(MyAppBar));
