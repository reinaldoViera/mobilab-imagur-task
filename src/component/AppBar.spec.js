import {
    mount
} from 'enzyme';
import React from 'react';
import { MyAppBar } from "./AppBar";
import renderer from 'react-test-renderer';

function setup() {
    return {
        history: {
            goBack: jest.fn()
        },
        location: {
            pathname: '/album/fAkEId'
        },
        classes: {

        }
    }
}

describe('AppBar component', () => {
    it('should show arrow back with album route', () => {
        const props = setup();
        const result = renderer.create( 
            <MyAppBar { ...props}/>
        ).toJSON();
        expect(result).toMatchSnapshot();
    });
    it('should go history back on arrow click', () => {
        const props = setup();
        const el = mount( <MyAppBar { ...props}/>);
        el.find('IconButton').simulate('click');
        expect(props.history.goBack.mock.calls.length).toEqual(1);        
    });
})