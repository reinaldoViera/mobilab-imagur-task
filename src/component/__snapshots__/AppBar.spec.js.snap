// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar component should show arrow back with album route 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      >
        Album
      </h6>
    </div>
  </header>
</div>
`;
